js/cptui.js

```diff
@@ -180,6 +180,22 @@ postboxes.add_postbox_toggles(pagenow);
 		return false;
 	});
 
+	$('#menu_icon').on('change', function() {
+		var value = $(this).val().trim();
+		$('#menu_icon_preview').html( composePreviewContent( value ) );
+	});
+
+	function composePreviewContent( value ) {
+		if ( ! value ) {
+			return '';
+		} else if ( 0 === value.indexOf('dashicons-') ) {
+			return $('<div class="dashicons-before"><br></div>').addClass( value );
+		}
+		else {
+			return $('<img />').attr( 'src', value );
+		}
+	}
```

Comment:
Kind of wish we didn't have to repeat quite so much with this spot, as we do in the admin ui class, but I understand why we have to.

Reply:
Sorry, I'm not sure I understood this phrase correctly. Unfortunately my English is not super :)

Should I fix something here or do you just agree with the inevitable "evil"?

Reply:
agreeing that it was inevitably evil in overall repeating of some code. Nothing to resolve here on your end.